### PR TITLE
Draft PR for IDS Data Transfer 

### PR DESCRIPTION
**Pull request by DominikPinsel**
Signed-off-by: Dominik Pinsel <dominik.pinsel@daimler.com>

### CODE REVIEW

File: extensions/api/control/src/main/java/org/eclipse/dataspaceconnector/api/control/ClientController.java [hunk 1]
```diff
@@ -18,48 +18,39 @@
...
-        DataRequest dataRequest = DataRequest.Builder.newInstance()
-                .dataDestination(dataAddress)
-                .connectorAddress(provider)
-                .connectorId(CONSUMER_ID)
-                .build();
         TransferProcess transferProcess = TransferProcess.Builder.newInstance()
-                .id(id)
+                .id(UUID.randomUUID().toString())
                 .dataRequest(dataRequest)
                 .build();
```
Comment (paullatzelsperger) [inline extensions/api/control/src/main/java/org/eclipse/dataspaceconnector/api/control/ClientController.java:56]: The correct way of doing this would be to call `TransferProcessManager.initiateConsumerRequest()`

File: data-protocols/ids/ids-api-multipart-endpoint-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/dispatcher/FutureCallback.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+//import org.jetbrains.annotations.NotNull;
+//
+//import java.io.IOException;
+//import java.util.concurrent.CompletableFuture;
+//import java.util.function.Function;
+//
+///**
+// * Shim from OK HTTP to the future returned by the ids dispatcher.
+// */
+//class FutureCallback<T> implements Callback {
```
Comment (paullatzelsperger) [inline data-protocols/ids/ids-api-multipart-endpoint-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/dispatcher/FutureCallback.java:29]: What's the point to checking in classes that entirely commented out?

File: data-protocols/ids/ids-api-multipart-endpoint-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/dispatcher/ArtifactRequestMessageSender.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+public class ArtifactRequestMessageSender implements IdsMessageSender<DataRequest> {
+    private final ObjectMapper objectMapper;
+    private final Monitor monitor;
+    private final OkHttpClient okHttpClient;
+    private final IdentityService identityService;
+    // TODO Create Container ID class to ensure its correctly build
+    private final String connectorId;
+    private final String connectorEndpoint;
+
+    public ArtifactRequestMessageSender(@NotNull Monitor monitor, @NotNull String connectorId, @NotNull String connectorEndpoint, @NotNull ObjectMapper objectMapper, @NotNull OkHttpClient okHttpClient, @NotNull IdentityService identityService) {
```
Comment (paullatzelsperger) [inline data-protocols/ids/ids-api-multipart-endpoint-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/dispatcher/ArtifactRequestMessageSender.java:44]: Does it make sense to have a message sender that can only ever send to one endpoint, i.e. the `connectorEndpoint`?

File: extensions/filesystem/configuration-fs/src/main/java/org/eclipse/dataspaceconnector/configuration/fs/FsConfigurationExtension.java [hunk 4]
```diff
@@ -69,6 +69,7 @@ public void initialize(Monitor monitor) {
             properties.load(is);
             for (String name : properties.stringPropertyNames()) {
                 propertyCache.put(name, properties.getProperty(name));
+                System.setProperty(name, properties.getProperty(name));
```
Comment (paullatzelsperger) [inline extensions/filesystem/configuration-fs/src/main/java/org/eclipse/dataspaceconnector/configuration/fs/FsConfigurationExtension.java:72]: Why is this necessary? seems like a hack.

File: launchers/ids-connector/vault.properties [hunk 5]
```diff
@@ -0,0 +1,2 @@
+consumer-blob-storage-key=r+lqbmhzLjLysUN/BnV3ZhOr0UXz1FT1MdWSCoXHIvUXrCp6zH9CYS5cfMlFoxRVwfXoogXNP3j4u4P7ZMQp3Q==
```
Comment (paullatzelsperger) [inline launchers/ids-connector/vault.properties:1]: You've just leaked your blob store key. NEVER check in sensitive information!!!

File: samples/other/copy-between-azure-and-s3/src/main/java/org/eclipse/dataspaceconnector/transfer/BlobToS3DataFlowController.java [hunk 6]
```diff
@@ -34,15 +34,15 @@ public BlobToS3DataFlowController(Vault vault, Monitor monitor, TypeManager type
     @Override
     public boolean canHandle(DataRequest dataRequest) {
```
Comment (paullatzelsperger) [inline samples/other/copy-between-azure-and-s3/src/main/java/org/eclipse/dataspaceconnector/transfer/BlobToS3DataFlowController.java:35]: this entire class exists in an almost identical form a second time in the `...extensions/transfer` package. I assume this is an oversight?

**Review by paullatzelsperger — COMMENTED** [review]
Just a few preliminary comments I had browsing through this. I assume much of it will get resolved once the PR reaches its final status.

File: data-protocols/ids/ids-api-multipart-endpoint-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/dispatcher/FutureCallback.java (hunk 2 above)
Comment (DominikPinsel) [reply data-protocols/ids/ids-api-multipart-endpoint-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/dispatcher/FutureCallback.java]: Draft PR :) still need to merge two more branches (one with my next commit) until this code becomes somewhat useable

File: data-protocols/ids/ids-api-multipart-endpoint-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/dispatcher/ArtifactRequestMessageSender.java (hunk 3 above)
Comment (DominikPinsel) [reply data-protocols/ids/ids-api-multipart-endpoint-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/dispatcher/ArtifactRequestMessageSender.java]: That's one of the draft implementations, I used to test my code. But it will be replaces by the ArtifactRequestMessage sender from #275

File: launchers/ids-connector/vault.properties (hunk 5 above)
Comment (DominikPinsel) [reply launchers/ids-connector/vault.properties]: Yea, totally forgot about this :)

File: samples/other/copy-between-azure-and-s3/src/main/java/org/eclipse/dataspaceconnector/transfer/BlobToS3DataFlowController.java (hunk 6 above)
Comment (DominikPinsel) [reply samples/other/copy-between-azure-and-s3/src/main/java/org/eclipse/dataspaceconnector/transfer/BlobToS3DataFlowController.java:35]: I will try to use sample 5 and removed all the `DataFlowController` code.

File: extensions/filesystem/configuration-fs/src/main/java/org/eclipse/dataspaceconnector/configuration/fs/FsConfigurationExtension.java (hunk 4 above)
Comment (DominikPinsel) [reply extensions/filesystem/configuration-fs/src/main/java/org/eclipse/dataspaceconnector/configuration/fs/FsConfigurationExtension.java]: This was necessary because the FsConfigurationExtension didn't work with the `propOrEnv` method. But as this is not necessary for this PR anymore, I removed the fix.

File: extensions/api/control/src/main/java/org/eclipse/dataspaceconnector/api/control/ClientController.java (hunk 1 above)
Comment (DominikPinsel) [reply extensions/api/control/src/main/java/org/eclipse/dataspaceconnector/api/control/ClientController.java:56]: Thanks for pointing it out. I updated the code :)

**Comment by jimmarino** [comment]
@DominikPinsel thanks! I'll merge.
